rotkehlchen/db/custom_assets.py

```diff
@@ -0,0 +1,100 @@
+import sqlite3
+from typing import TYPE_CHECKING, List, Tuple
+
+from rotkehlchen.assets.asset import CustomAsset
+from rotkehlchen.assets.types import AssetType
+from rotkehlchen.db.filtering import CustomAssetsFilterQuery
+from rotkehlchen.errors.misc import InputError
+from rotkehlchen.globaldb import GlobalDBHandler
+
+if TYPE_CHECKING:
+    from rotkehlchen.db.dbhandler import DBHandler
+    from rotkehlchen.db.drivers.gevent import DBCursor
+
+
+class DBCustomAsset:
+    def __init__(self, db_handler: 'DBHandler') -> None:
+        self.db = db_handler
+        self.global_db = GlobalDBHandler()
+
+    def _get_custom_assets(
+            self,
+            cursor: 'DBCursor',
+            filter_query: CustomAssetsFilterQuery,
+    ) -> List[CustomAsset]:
+        query, bindings = filter_query.prepare()
+        query = 'SELECT identifier, name, notes, type, started, swapped_for FROM custom_assets ' + query  # noqa: E501
+        cursor.execute(query, bindings)
+        return [CustomAsset.deserialize_from_db(entry) for entry in cursor]
+
+    def get_custom_assets_and_limit_info(
+            self,
+            cursor: 'DBCursor',
+            filter_query: CustomAssetsFilterQuery,
+    ) -> Tuple[List[CustomAsset], int]:
+        """Retrieves all custom assets with pagination."""
+        custom_assets = self._get_custom_assets(
+            cursor=cursor,
+            filter_query=filter_query,
+        )
+        query, bindings = filter_query.prepare(with_pagination=False)
+        query = 'SELECT COUNT(*) from custom_assets ' + query
+        total_found_result = cursor.execute(query, bindings).fetchone()[0]
+        return custom_assets, total_found_result
```

Comment:
You are already querying the total found, just return it as we do in other endpoints